File: apps/api/src/api/routes/operations.py
from fastapi import APIRouter

from src.core.runtime import get_runtime_container

router = APIRouter(tags=['operations'])


@router.get('/tasks')
def list_tasks() -> dict:
    container = get_runtime_container()
    return {'items': container.metadata.list_tasks()}


@router.get('/runs')
def list_runs() -> dict:
    container = get_runtime_container()
    return {'items': container.metadata.list_job_runs()}


@router.get('/exports')
def list_exports() -> dict:
    container = get_runtime_container()
    return {'items': container.metadata.list_export_jobs()}

File: apps/bff/src/services/operations.ts
import type { ExportItem, RunItem, TaskItem } from '../types.js'
import { platformFetch } from './platform.js'

export async function listTasks() {
  const response = (await platformFetch('/tasks')) as { items?: TaskItem[] }
  return response.items ?? []
}

export async function listRuns() {
  const response = (await platformFetch('/runs')) as { items?: RunItem[] }
  return response.items ?? []
}

export async function listExports() {
  const response = (await platformFetch('/exports')) as { items?: ExportItem[] }
  return response.items ?? []
}